core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -206,16 +206,16 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
         val container = pr.getUserAttribute.asInstanceOf[GemFireContainer]
         if (ColumnFormatRelation.isColumnTable(table) &&
             pr.getLocalMaxMemory > 0) {
-          // TODO: this should use a transactional iterator to get a consistent
-          // snapshot (also pass the same transaction to getNumColumnsInTable
-          //   for reading value and delete count)
-          val itr = pr.localEntriesIterator(null.asInstanceOf[InternalRegionFunctionContext],
-            true, false, true, null).asInstanceOf[PartitionedRegion#PRLocalScanIterator]
           var numColumnsInTable = -1
           // Resetting PR numRows in cached batch as this will be calculated every time.
           var rowsInColumnBatch = 0L
           var offHeapSize = 0L
           if (container ne null) {
+            // TODO: this should use a transactional iterator to get a consistent
```

Comment:
Shifting inside ne null is good. But this is by inspection or is it fixing some reported problem? and shouldn't he container be always non null?

Reply:
Right null check looks redundant. However, the main change here is not the shifting but the "includeValues" parameter pr.localEntriesIterator which is now false. So stats iteration will always use map iteration only and not attempt to sort by disk entries etc (stats rows should always be in-memory).